solutions/instances/main.tf

```diff
@@ -38,6 +38,26 @@ locals {
   scc_workload_protection_instance_name     = var.prefix != null ? "${var.prefix}-${var.scc_workload_protection_instance_name}" : var.scc_workload_protection_instance_name
   scc_workload_protection_resource_key_name = var.prefix != null ? "${var.prefix}-${var.scc_workload_protection_instance_name}-key" : "${var.scc_workload_protection_instance_name}-key"
   scc_cos_bucket_name                       = var.prefix != null ? "${var.prefix}-${var.scc_cos_bucket_name}" : var.scc_cos_bucket_name
+  create_cross_account_auth_policy          = nonsensitive(!var.skip_cos_kms_auth_policy && var.ibmcloud_kms_api_key != null)
+
+  kms_service_name = var.existing_kms_instance_crn != null ? (
```

Comment:
I think the CRN parser could be used for this too now?

Reply:
I think Steve found a bug with CRN parser that prevents it being used here currently, so for now, proceed with this approach